C/plugins/storage/sqlite/common/readings.cpp

```diff
@@ -820,11 +820,14 @@ int blocks = 0;
 			{
 				// search in earlier/left half
 				r = m - 1;
+
+				// The m position should be skipped as midRowId is 0
+				m = r;
```

Comment:
'm' position is being skipped anyway, since for next iteration, we are setting right end as 'm-1' here. And then at beginning of next iteration, 'm' would be recalculated as middle of range (l..r).

Reply:
the assignment is for handling the case in which it is the very last row that should be skipped as it is newer than age, actually the test id 102.